get_dependencies.py

```diff
@@ -23,19 +23,39 @@ def download(url, filename):
                 sys.stdout.flush()
     sys.stdout.write('\n')
 
-
-print('[*] Extracting webdriver archives...')
-for f in ['geckodriver', 'nightly', 'chromedriver', 'chrome-linux']:
-    tar = tarfile.open('tools/%s.tar.xz' % f, 'r:xz')
-    tar.extractall(path='tools/')
-    tar.close()
+if sys.platform == "linux":
+    print('[*] Downloading support files for linux ~ 108 MB')
+    download('https://www.dropbox.com/s/3mosdy9tsf7pilh/linux.zip?dl=1', 'linux.zip')
+    print('[*] Extracting linux.zip...')
+    with ZipFile('linux.zip', 'r') as z:
+        z.extractall("linux")
+    print('[*] Extracting webdriver archives...')
+    for f in ['geckodriver', 'nightly', 'chromedriver', 'chrome-linux']:
+        tar = tarfile.open('linux/%s.tar.xz' % f, 'r:xz')
+        tar.extractall(path='tools/')
+        tar.close()
+    os.remove('linux.zip')
+    shutil.rmtree('linux')
+
+elif sys.platform == "darwin":
+    print('[*] Downloading support files for mac ~ 150 MB')
+    download('https://www.dropbox.com/s/k1szymmktj0e1pf/mac.zip?dl=1', 'mac.zip')
+    print('[*] Extracting mac.zip...')
+    os.system("unzip mac.zip -d tools")
```

Comment:
We shouldn't use `os.system` here, but the standard ZipFile class. If the extracted executable files can't be run, it's probably just a permission issue that you can solve by setting the executable bits.

Reply:
all the files are getting corrupt i am not able to figure out.

Reply:
Try extracting them with ZipFile, then go in the directory where they are extracted, run `ls -al` and post the result here.

Reply:
ls -al
total 30288
drwxr-xr-x   8 amit  staff       256 Feb  7 00:19 .
drwxr-xr-x  28 amit  staff       896 Feb  7 00:18 ..
-rw-r--r--@  1 amit  staff      6148 Feb  7 00:17 .DS_Store
drwxr-xr-x   3 amit  staff        96 Feb  7 00:19 Nightly.app
drwxr-xr-x   8 amit  staff       256 Feb  7 00:19 __MACOSX
drwxr-xr-x   3 amit  staff        96 Feb  7 00:19 chrome.app
-rw-r--r--   1 amit  staff  11893548 Feb  7 00:19 chromedriver
-rw-r--r--   1 amit  staff   3600548 Feb  7 00:19 geckodriver

Reply:
when i try to open the chrome  then a message is displayed : You can’t open the application “chrome” because it may be damaged or incomplete.

Reply:
`geckodriver` and `chromedriver` are not executable, that's probably why you can't run them.

Reply:
Check if `chrome` has the same problem of permissions.

Reply:
-rw-r--r--  1 amit  staff  19072 Feb  7 00:19 chrome

Reply:
It does.

Reply:
any solution to this?

Reply:
Told you in the first comment 😄 
Try to set the files as executable.

Reply:
okay 😄